osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -444,10 +444,18 @@ Task IMultiplayerClient.HostChanged(int userId)
             return Task.CompletedTask;
         }
 
-        Task IMultiplayerClient.SettingsChanged(MultiplayerRoomSettings newSettings)
+        async Task IMultiplayerClient.SettingsChanged(MultiplayerRoomSettings newSettings)
         {
+            Debug.Assert(APIRoom != null);
+            Debug.Assert(Room != null);
+
+            // ensure the new selected item is populated immediately.
+            var playlistItem = APIRoom.Playlist.SingleOrDefault(p => p.ID == newSettings.PlaylistItemId);
```

Comment:
Somehow, this needs to be scheduled. It's on the signalr thread.

Reply:
Which part is the issue? The access to `APIRoom`?

Reply:
Correct